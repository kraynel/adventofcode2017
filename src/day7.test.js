const { getRootNode } = require("./day7");
const fs = require("fs");

describe("getRootNode", () => {
  it("should get the root node", () => {
    expect(
      getRootNode(`pbga (66)
xhth (57)
ebii (61)
havc (66)
ktlj (57)
fwft (72) -> ktlj, cntj, xhth
qoyq (66)
padx (45) -> pbga, havc, qoyq
tknk (41) -> ugml, padx, fwft
jptl (61)
ugml (68) -> gyxo, ebii, jptl
gyxo (61)
cntj (57)`)
    ).toBe("tknk");
  });

  it("should get the value", done => {
    fs.readFile("inputs/day7.txt", "utf8", (err, data) => {
      expect(getRootNode(data)).toMatchSnapshot();
      done();
    });
  });
});
