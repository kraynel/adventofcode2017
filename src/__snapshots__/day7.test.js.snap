// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`balanceTree should get the value 1`] = `1206`;

exports[`getRootNode should get the value 1`] = `"airlri"`;
